### DISCUSSION TITLE
Equality checking

### DISCUSSION THREAD

**Discussion by zmitchell** [toml-rs/toml#656] [Q&A]
I wanted to implement a check to see if a TOML file has changed in a semantically meaningful way, so I went to compare two parsed `Document`s, but it looks like there's no implementation of `Eq` or `PartialEq` for most of the structurally significant types. I'm assuming there's a good reason for this.

Is it "wrong" or complicated to check equality of these types? Should I be using a different crate to do this?

**Comment by epage** [comment]
The question is what to compare. For a document, to literally compare would be to check insignificant details. Now `toml::Table` would make sense to implement `PartalEQ` if it doesn't already.
